instructor/common.py

```diff
@@ -415,81 +424,110 @@ def assignments_edit_treatments(request):
+                                     can_delete=True)
     if request.method == "POST":
-        formset = TreatmentsFormSet(request.POST)
+        formset = STFormSet(request.POST)
         formset.clean()
         if formset.is_valid():
-            message = "Thank you"
-            for form in formset.ordered_forms:
-                form.instance.order = form.cleaned_data['ORDER']
             entries = formset.save(commit=False)
+            for obj in formset.deleted_objects:
+                obj.delete()
             for form in entries:
-                form.protocol = protocol
                 form.save()
-        else:
-            message = "Something went wrong"
+    else:
+        # If there are no Treatments want to create them
+        if not models.Treatment.objects.filter(assignment=assignment).exists():
+            create_treatments(assignment)
+            create_strain_treatments(assignment)
 
-    formset = TreatmentsFormSet(queryset=models.Treatments.objects.filter(protocol=protocol).order_by('order'))
-    return render_to_response('instructor/treatments.html',
+    formset = STFormSet(queryset=models.StrainTreatment.objects.filter(assignment=assignment))
+    return render_to_response('instructor/strain_protocols.html',
                               {'formset': formset,
-                               'message': message,
-                               'assignment': assignment
+                               'assignment_name': assignment.name,
+                               'headers': headers,
+                               'has_concentration': assignment.has_concentration,
+                               'has_temperature': assignment.has_temperature,
+                               'has_start_time': assignment.has_start_time,
+                               'has_duration': assignment.has_duration,
+                               'has_collection_time': assignment.has_collection_time
                               },
                               context_instance=RequestContext(request))
 
 
-def strain_treatments_edit(request, assignment):
-    assignment = models.Assignment.objects.get(id=assignment)
+def create_treatments(assignment):
+    drugs = models.Drug.objects.filter(assignment=assignment)
+    temperature = models.Temperature.objects.filter(assignment=assignment)
+    collection_times = models.CollectionTime.objects.filter(assignment=assignment)
+    # Creating treatments
+    if assignment.has_temperature and assignment.has_collection_time:
+        for d in drugs:
+            for t in temperature:
+                for c in collection_times:
+                    treatment, created = models.Treatment.objects.get_or_create(
+                        drug=d,
+                        temperature=t,
+                        collection_time=c,
+                        assignment=assignment)
+                    treatment.save()
```

Comment:
I think the `.save()` here is unnecessary since it's done as an insert within `get_or_create`